lib/vagrant-azure/config.rb

```diff
@@ -147,6 +151,23 @@ def validate(machine)
           @private_key_file = File.expand_path(paths.first)
         end
 
+        if !@data_disks.nil? && @data_disks != UNSET_VALUE
+          data_disk_example = "{disk_label: 'disk1', disk_size: 40}"
+          unless @data_disks.is_a? Array
+            machine.ui.warn("data_disks is expected to be defined with an Array.\nFor instance [#{data_disk_example}, ...].")
+            @data_disks = nil
+          end
+          @data_disks.each do |dict|
+            unless dict.is_a? Hash
```

Comment:
If this is not a hash, perhaps add an error to the errors. I don't think we'd want to continue past configuration in this scenario. Possibly for the keys as well if they are required.